tubesync/sync/views.py

```diff
@@ -296,23 +296,30 @@ class EditSourceMixin:
     def form_valid(self, form: Form):
         # Perform extra validation to make sure the media_format is valid
         obj = form.save(commit=False)
-        source_type = form.cleaned_data['media_format']
+        # temporarily use media_format from the form
+        saved_media_format = obj.media_format
+        obj.media_format = form.cleaned_data['media_format']
         example_media_file = obj.get_example_media_format()
+        obj.media_format = saved_media_format
 
-        if example_media_file == '':
+        if '' == example_media_file:
             form.add_error(
                 'media_format',
                 ValidationError(self.errors['invalid_media_format'])
             )
 
         # Check for suspicious file path(s)
         try:
-            targetCheck = form.cleaned_data['directory']+"/.virt"
-            newdir = safe_join(settings.DOWNLOAD_ROOT,targetCheck)
+            targetCheck = form.cleaned_data['directory'] + '/.virt'
+            safe_join(settings.DOWNLOAD_ROOT, targetCheck)
         except SuspiciousFileOperation:
```

Comment:
Using the exception to check for invalid values was cute, but not intuitive or obvious.